### PR TITLE
Add chip-tool test suite

### PR DESCRIPTION
**Pull request by MonicaisHer**
```
matter-snap-testing/test/suites/chip-tool$ go test -v -failfast -count 1 .
2024/01/03 10:18:57 [CLEAN]
2024/01/03 10:18:57 [exec] sudo snap remove --purge chip-tool
[sudo] password for mengyi: 
2024/01/03 10:19:03 [stdout] chip-tool removed
2024/01/03 10:19:03 [SETUP]
2024/01/03 10:19:03 [exec] sudo snap install chip-tool --channel=latest/edge
2024/01/03 10:19:05 [stdout] chip-tool (edge) v1.2.0.1+snap from Canonical IoT Labs installed
2024/01/03 10:19:05 [exec] sudo snap connect chip-tool:avahi-observe 
2024/01/03 10:19:05 [exec] sudo snap connect chip-tool:bluez 
2024/01/03 10:19:06 [exec] sudo snap connect chip-tool:process-control 
=== RUN   TestMatterDeviceOperations
=== RUN   TestMatterDeviceOperations/Commission
    exec.go:19: [exec] sudo chip-tool pairing onnetwork 110 20202021
=== RUN   TestMatterDeviceOperations/Control
    exec.go:19: [exec] sudo chip-tool onoff toggle 110 1
    main_test.go:115: Retry 1/10: Waiting for expected content in logs: CHIP:ZCL: Toggle ep1 on/off
    main_test.go:124: Found expected content in logs: CHIP:ZCL: Toggle ep1 on/off
--- PASS: TestMatterDeviceOperations (1.85s)
    --- PASS: TestMatterDeviceOperations/Commission (0.53s)
    --- PASS: TestMatterDeviceOperations/Control (1.31s)
PASS
2024/01/03 10:19:08 [TEARDOWN]
2024/01/03 10:19:08 [exec] (sudo journalctl --since "2024-01-03 10:18:57" --no-pager | grep "chip-tool"|| true) > chip-tool.log
Wrote snap logs to /home/mengyi/Desktop/881/matter-snap-testing/test/suites/chip-tool/chip-tool.log
2024/01/03 10:19:08 Removing installed snap: true
2024/01/03 10:19:08 [exec] sudo snap remove --purge chip-tool
2024/01/03 10:19:10 [stdout] chip-tool removed
ok      matter-snap-testing/test/suites/chip-tool       13.659s
```

### CODE REVIEW

File: test/suites/chip-tool/chip-all-clusters-minimal-app
Comment (MonicaisHer) [inline test/suites/chip-tool/chip-all-clusters-minimal-app:1]: Replace with artifact

**Comment by MonicaisHer** [comment]
Could you please review it @farshidtz?

File: build/action.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    # Get snap base version
+    - shell: bash
+      run: |
+        echo "base=$(sed -n 's/^base: //p' snap/snapcraft.yaml)" >> $GITHUB_ENV
+
+    # Build the snap
+    - uses: snapcore/action-build@v1
+      id: build
+      with:
+        snapcraft-args: ${{ env.base == 'core22' && '--verbose' || '' }}
```
Comment (farshidtz) [inline build/action.yml:41]: Checking the base to toggle verbosity is no longer required; see https://bugs.launchpad.net/snapcraft/+bug/1980515

**Review by farshidtz — CHANGES_REQUESTED** [review]
When ran locally, the test run successfully only once. After that, chip tool fails to commission with a timeout error. Need to clearing temporary chip files to prevent that.

The change history needs to be improved before merging. In particular, the commit that copies code from [edgex-snap-testing](https://github.com/canonical/edgex-snap-testing) needs to have original [co-authors](https://docs.github.com/en/pull-requests/committing-changes-to-your-project/creating-and-editing-commits/creating-a-commit-with-multiple-authors#creating-co-authored-commits-on-the-command-line) and more details on the origin. It is best to fork in such situations instead of copy-pasting the code. I also noticed that the original code did not have a license. 

Please refer to other comments inline.

File: test/suites/chip-tool/main_test.go [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+	}
+
+	code := m.Run()
+	teardown()
+
+	os.Exit(code)
+}
+
+
+var cmd *exec.Cmd
```
[the remark this answers is not in the document]
Comment (MonicaisHer) [reply test/suites/chip-tool/main_test.go:32]: This variable has been initialized outside of the function, so the cluster app instance will be destroyed after the subtest finishes.

**Comment by farshidtz** [comment]
> This PR has been migrated to a new [PR](https://github.com/MonicaisHer/matter-snap-testing/pull/5). Please see that all review changes have been applied there.

Please sync the changes on main and IENG-881 branches of your new repo back to this repo (rewrite the history by force pushing). Creating an entirely new repo wasn't needed, as that means we lose this PR and reviews.

File: test/suites/chip-tool/main_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (farshidtz) [reply test/suites/chip-tool/main_test.go:32]: > This variable has been initialized outside of the function, so the cluster app instance will be destroyed after the subtest finishes.

I'm not sure what you mean. The `cmd` variable is **declared globally** and **initialized inside the function**. How is the global declaration of this variable associated with destroying the cluster app instance?

**Comment by MonicaisHer** [comment]
> > This PR has been migrated to a new [PR](https://github.com/MonicaisHer/matter-snap-testing/pull/5). Please see that all review changes have been applied there.
> 
> Please sync the changes on main and IENG-881 branches of your new repo back to this repo (rewrite the history by force pushing). Creating an entirely new repo wasn't needed, as that means we lose this PR and reviews.

The changes have been synced to the main branch, and the commit history has been rewritten on IENG-881 by force pushing. However, after force-pushing, this PR has been automatically closed and can't be reopened. 

Would you like me to open a new PR on this repository and reference this closed PR to maintain a trace of reviews?

**Comment by farshidtz** [comment]
> The changes have been synced to the main branch, and the commit history has been rewritten on IENG-881 by force pushing. However, after force-pushing, this PR has been automatically closed and can't be reopened.
> 

Something isn't right. 

The history from [IENG-881 branch](https://github.com/canonical/matter-snap-testing/tree/IENG-881) on the new repo:
```
$ git remote -v
origin	https://github.com/canonical/matter-snap-testing.git (fetch)
origin	https://github.com/canonical/matter-snap-testing.git (push)
$ git log --oneline
c747970 (HEAD -> IENG-881, origin/IENG-881) Update Git LFS attribute
5c7869c Remove go routine, remove edgex related utils
940edac Add Git LFS tracking for large binary file
62e3996 Add license
0cb0823 Remove edgex related code, add matter test suite
4ec7094 (origin/main, origin/HEAD, main) Improve input descriptions in snap-testing.yml (#222)
```

The history from IENG-881 on this repo:
```
$ git remote -v
origin	https://github.com/MonicaisHer/matter-snap-testing.git (fetch)
origin	https://github.com/MonicaisHer/matter-snap-testing.git (push)
$ git log --oneline
d75aaf9 (HEAD -> IENG-881, origin/IENG-881) Remove edgex test suites, add matter chip-tool test suite
4ec7094 (origin/main, origin/HEAD, main) Improve input descriptions in snap-testing.yml (#222)
a85ee86 Add channel input for Github snap testing workflow (#221)
```

And then there is this PR's diff showing something completely different. Github can't produce a diff:
image


> Would you like me to open a new PR on this repository and reference this closed PR to maintain a trace of reviews?

If you can't resolve the issue, then yes that would be the second-best option.

**Pull request closed without merging**